src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -416,28 +415,135 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+
+    # Get the access credentials for the registry
+    registry_name, username, password = get_access_credentials(
+        cmd,
+        registry_name=registry_name,
+        tenant_suffix=None,
+        username=None,
+        password=None,
+        repository=repository_name,
+        permission='pull')
+    
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+    
+    response = requests.get(
+        manifest_url,
+        headers={'Accept': 'application/vnd.oci.artifact.manifest.v1+json'
+            ', application/vnd.cncf.oras.artifact.manifest.v1+json'
+            ', application/vnd.oci.image.manifest.v1+json'
+            ', application/vnd.oci.image.index.v1+json'
+            ', application/vnd.docker.distribution.manifest.v2+json'
+            ', application/vnd.docker.distribution.manifest.list.v2+json'},
+        auth=(username, password))
+    
+    manifest = response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    def get_smallest_blob_digest(manifest):
+       
+        smallest_blob = min(manifest['layers'], key=lambda layer: layer['size'])
+        smallest_blob_digest = smallest_blob['digest']
+
+        return smallest_blob_digest
+
+    digest = get_smallest_blob_digest(manifest)
+
+    # Pull blob - GET {url}/v2/{name}/blobs/{digest}
+    request_url = 'https://' + login_server + '/v2/' + repository_name + '/blobs/' + digest
+    
+    logger.debug(add_timestamp("Sending a HTTP GET request to {}".format(request_url)))
+    
+    response = requests.get(request_url, auth=(username, password))
+    
+    if response.status_code < 400:
+        print_pass("Blob identified by image '{}' is available to pull from registry '{}'".format(image, login_server))
+    else:
+        raise CLIError("'Blob identified by image '{}' cannot be pulled from registry '{}'".format(image, login_server)) 
 
 # General command
 def acr_check_health(cmd,  # pylint: disable useless-return
                      vnet=None,
                      ignore_errors=False,
                      yes=False,
-                     registry_name=None):
+                     registry_name=None,
+                     repository_name=None,
+                     image=None):
     from azure.cli.core.util import in_cloud_console
+
+    # If repository and image are provided, validate them first before running other checks
+    if (image and repository_name):
+        validate_repository_and_image(cmd, registry_name, repository_name, image)
+
     in_cloud_console = in_cloud_console()
     if in_cloud_console:
         logger.warning("Environment checks are not supported in Azure Cloud Shell.")
     else:
         _get_docker_status_and_version(ignore_errors, yes)
         _get_cli_version()
-
+    
     _check_registry_health(cmd, registry_name, ignore_errors)
 
+    if image and repository_name:
```

Comment:
Why do we do validate repository and image before check registry health and then check blob exists after? Intuitively I would think you'd want to check them both back to back, I would think after checking registry health since if there's a network issue or something then you definitely won't be able to hit the repo

Reply:
Yes, good point, I changed the ordering of these methods so check_registry_health is first.